src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -76,6 +75,22 @@ void VehicleInfoPlugin::OnApplicationEvent(
   if (plugins::ApplicationEvent::kApplicationRegistered == event) {
     application->AddExtension(
         std::make_shared<VehicleInfoAppExtension>(*this, *application));
+  } else if (plugins::ApplicationEvent::kApplicationUnregistered == event) {
+    DCHECK_OR_RETURN_VOID(application);
```

Comment:
@IGapchuk extract logic to a function

Reply:
Done in 73c67bf4f764f384b64cc45f9a7a3087eff31f1f.

Reply:
Best to put this dcheck in the function as well.